consensus/block_validation.go

```diff
@@ -46,29 +47,44 @@ type MessageSyntaxValidator interface {
 type DefaultBlockValidator struct {
 	clock.Clock
 	blockTime time.Duration
+	ptv       *version.ProtocolVersionTable
 }
 
 // NewDefaultBlockValidator returns a new DefaultBlockValidator. It uses `blkTime`
 // to validate blocks and uses the DefaultBlockValidationClock.
-func NewDefaultBlockValidator(blkTime time.Duration, c clock.Clock) *DefaultBlockValidator {
+func NewDefaultBlockValidator(blkTime time.Duration, c clock.Clock, ptv *version.ProtocolVersionTable) *DefaultBlockValidator {
 	return &DefaultBlockValidator{
 		Clock:     c,
 		blockTime: blkTime,
+		ptv:       ptv,
 	}
 }
 
 // ValidateSemantic validates a block is correctly derived from its parent.
-func (dv *DefaultBlockValidator) ValidateSemantic(ctx context.Context, child *types.Block, parents *types.TipSet) error {
+func (dv *DefaultBlockValidator) ValidateSemantic(ctx context.Context, child *types.Block, parents *types.TipSet, parentWeight uint64) error {
 	pmin, err := parents.MinTimestamp()
 	if err != nil {
 		return err
 	}
 
+	// Alphanet upgrade
 	ph, err := parents.Height()
 	if err != nil {
 		return err
 	}
 
+	parentVersion, err := dv.ptv.VersionAt(types.NewBlockHeight(ph))
+	if err != nil {
+		return err
+	}
+	if parentVersion >= version.Protocol1 {
```

Comment:
Add a significant comment here explaining the old and new behaviour, and what has changed.